### ISSUE TITLE
[Epic] Rancher Component Library

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#5108]
**Detailed Description**

Separate a group of Rancher Components that can be used as a component library. 

These are the components & utils that have been identified as a good starting point for the first iteration:

```
Components
  |- BadgeState.vue
  |- Banner.vue
  |- Card.vue
  |- Form
    |- Checkbox.vue
    |- LabeledInput.vue
    |- LabeledTooltip.vue
    |- RadioButton.vue
    |- RadioGroup.vue
    |- TextAreaAutoGrow.vue
```
```
Utils
  |- array.js
  |- object.js
  |- platform.js
  |- position.js
  |- sort.js
  |- string.js
  |- unit.js 
```

**Context**

We have several projects that are starting to use shared components throughout Rancher. For example, Rancher Desktop is currently copying dashboard components into their repo. We can reduce code duplication a great deal with a component framework.

### TODO

- [x] rancher/dashboard#5932
- [x] rancher/dashboard#6088
- [x] rancher/dashboard#5934
- [x] rancher/dashboard#5930
- [ ] rancher/dashboard#5933
- [ ] rancher/dashboard#5935
- [ ] rancher/dashboard#5929
- [x] rancher/dashboard#5931
- [ ] rancher/dashboard#5996
- [ ] rancher/dashboard#6005
- [x] rancher/dashboard#6649
- [x] rancher/dashboard#6655
- [x] rancher/dashboard#6741
- [ ] rancher/dashboard#6759
- [ ] rancher/dashboard#6760
- [x] rancher/dashboard#6787

**Comment by rak-phillip** [comment]
`ActionMenu.vue` looks like it is a dependency of `SortableTable` will need to be extracted later, if at all, as it will require quite a bit of refactoring. Dealing wit `SortableTable` will take a considerable amount of effort and time, removing from this initial pass.

**Comment by nwmac** [comment]
@rak-phillip Can we use this as the issue for 2.16 work on component library - probably needs reshaping, but want to have a clear objective for 2.16

**Comment by rak-phillip** [comment]
@nwmac certainly, we can repurpose this Epic to contain our 2.16 objectives. 

FYI - we have our current objectives detailed in Rancher Components project board: https://github.com/orgs/rancher/projects/57/views/66, but I don't think it hurts to reshape this epic and list them here as well.
